python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -1169,8 +1198,25 @@ def do_evaluate(
             raise NotImplementedError(
                 f"Agg in context {context}"
             )  # pragma: no cover; unreachable
-        (child,) = self.children
-        return self.op(child.evaluate(df, context=context, mapping=mapping))
+
+        # Don't convert scalar literals to pylibcudf column
+        # for quantile (AggInfo takes in Python scalars not pylibcudf ones)
+        if self.name == "quantile":
+            evaled_children = []
+            for child in self.children:
+                if isinstance(child, Literal):
+                    evaled_children.append(child.value)
+                else:
+                    evaled_children.append(
+                        child.evaluate(df, context=context, mapping=mapping)
+                    )
+        else:
+            evaled_children = [
+                child.evaluate(df, context=context, mapping=mapping)
+                for child in self.children
+            ]
+
+        return self.op(*evaled_children)
```

Comment:
Then this can be:
```
if self.name == "quantile":
    child, _ = self.children
else:
    (child,) = self.children
return self.op(child.evaluate(df, context=context, mapping=mapping))
```